### PR TITLE
Doc debian 10 updates

### PR DESCRIPTION
**Pull request by robje**
Some (Debian specific) updates to the Debian 10 installation procedure.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by SourceDoctor** [comment]
some questions:

- changing from nano to editor? why?
- chmod on /etc/snmp/snmpd.conf  why?
- what do you want to do with dpkg-divert --rename /etc/snmp/snmpd.conf
- nginx configuration -> as i know it's a *.conf not a *.vhost
- why remove this line: rm /etc/nginx/sites-available/default && rm /etc/nginx/sites-enabled/default

**Comment by robje** [comment]
> some questions:

apparently I need more git school. I thought I answered most of these questions in the commit messages. Apparently my additional lines went missing.

>     * changing from nano to editor? why?

Debian has the command editor which is the users preferred editor. It can be nano or emacs or vim or something different.
Using editor here also better indicates the user is free to chose an editor.

>     * chmod on /etc/snmp/snmpd.conf  why?

snmpd.conf holds (can hold) sensitive information like community stings and passwords.
This is why the file by default is mode 600 on installation. The chmod is to restore this after copying the librenms version.

>     * what do you want to do with dpkg-divert --rename /etc/snmp/snmpd.conf

This is to let the package manager know this file is now managed in an other way than through the package manager.
It results in less user interaction on upgrades.

>     * nginx configuration -> as i know it's a *.conf not a *.vhost

This was not changed. The file was called *.vhost already. And it works as the include is *
```nginx
include /etc/nginx/sites-enabled/*;
```

>     * why remove this line: rm /etc/nginx/sites-available/default && rm /etc/nginx/sites-enabled/default

This line is not removed.
I've updated it to remove the sites-enabled file only. That way the package manager can still manage the sites-available version.

**Comment by SourceDoctor** [comment]
**changing editor from `nano` to `editor `**
may be good, but i would it keep on `nano`, or change it to `editor `on all debian/ubuntu system documentations, but this should be a seperate PR

*snmpd chmod*
if you do it, you have to be ensure file is owned bei snmpd process.
So a chown with user and group is missing

**dpkg-divert**
not exists on any documentations i've ever seen, also not here.
To avoid user confusion i would keep it away

**Comment by SourceDoctor** [comment]
@robje something new about it?

**Comment by murrant** [comment]
@robje maybe a note about editor.  So people don't get stuck in vim ;D

Also, this needs merging with  upstream.

**Comment by SourceDoctor** [comment]
something new here?

**Comment by robje** [comment]
> @robje maybe a note about editor. So people don't get stuck in vim ;D

AFAIK nano is the default editor in Debian, no worries about getting stuck in vim unless one really wants to :)

> Also, this needs merging with upstream.

- [x] done

**Comment by robje** [comment]
> **changing editor from `nano` to `editor `**
> may be good, but i would it keep on `nano`, or change it to `editor` on all debian/ubuntu system documentations, but this should be a separate PR

agreed

> **snmpd chmod**
> if you do it, you have to be ensure file is owned bei snmpd process.
> So a chown with user and group is missing

snmpd.conf is read before dropping privileges. On a new installation it is owned by root.

> **dpkg-divert**
> not exists on any documentations i've ever seen, also not here.
> To avoid user confusion i would keep it away

Will put this forward in a separate pull request

File: doc/Installation/Installation-Debian-10-Nginx.md [hunk 1]
```diff
@@ -134,13 +134,15 @@ server {
...
-systemctl reload nginx
+ln -s /etc/nginx/sites-available/librenms.vhost /etc/nginx/sites-enabled/librenms.vhost
+service nginx reload
 ```
 
 # Configure snmpd
 
 ```bash
 cp /opt/librenms/snmpd.conf.example /etc/snmp/snmpd.conf
+chmod 600 /etc/snmp/snmpd.conf
```
Comment (SourceDoctor) [inline doc/Installation/Installation-Debian-10-Nginx.md:145]: you change mod to 600 ... doesn't work if snmpd.conf is not owned by snmpd user

Comment (SourceDoctor) [reply doc/Installation/Installation-Debian-10-Nginx.md:145]: if you limit access such high a chown own is also needed

Comment (robje) [reply doc/Installation/Installation-Debian-10-Nginx.md:145]: It is not. snmpd is started as root and privileges are dropped after starting. snmpd must be started as root to acquire privileged port 161.

snmpd.conf is read as root, before dropping privileges.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-65-release-changelog-june-2020/12687/1

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-65-release-changelog-june-2020/12687/1

**Pull request merged**